api/auth_middleware_test.go

```diff
@@ -8,24 +8,48 @@ import (
 	helpers "github.com/topfreegames/Will.IAM/testing"
 )
 
-func TestAuthMiddlewareKeyPair(t *testing.T) {
+func TestAuthMiddlewareKeyPairShouldAuthenticateUser(t *testing.T) {
 	helpers.CleanupPG(t)
 
-	rootSA := helpers.CreateRootServiceAccount(t)
-
+	rootSA := helpers.CreateRootServiceAccountWithKeyPair(t)
 	app := helpers.GetApp(t)
-
 	req, _ := http.NewRequest(http.MethodGet, "/service_accounts", nil)
+
 	req.Header.Set("Authorization", fmt.Sprintf(
-		"KeyPair %s:%s", rootSA.KeyID, rootSA.KeySecret,
+		"keypair %s:%s", rootSA.KeyID, rootSA.KeySecret,
 	))
-	rec := helpers.DoRequest(t, req, app.GetRouter())
-	if rec.Code != http.StatusOK {
-		t.Errorf("Expected status %d. Got %d", http.StatusOK, rec.Code)
+
+	response := helpers.DoRequest(t, req, app.GetRouter())
+
+	if response.Code != http.StatusOK {
+		t.Errorf("Expected status %d. Got %d", http.StatusOK, response.Code)
 	}
 
-	serviceAccountName := rec.Header().Get("x-service-account-name")
+	serviceAccountName := response.Header().Get("x-service-account-name")
 	if serviceAccountName != "root" {
 		t.Errorf("Expected service account name %s. Got %s", "root", serviceAccountName)
 	}
 }
+
+func TestAuthMiddlewareBearerShouldAuthenticateUser(t *testing.T) {
```

Comment:
We are not testing failure scenarios here yet. For instance, testing when an unknown authorisation type is sent `like Basic` or if the Bearer token or the key pair is invalid... I'm not sure if I should add them in this PR - because of its size - or in the following ones 🤔 💭 . Thoughts?

Reply:
either way works